File: 02-06-2024_Construct_list_using_given_q_XOR_queries.py
class Solution:
    def constructList(self, k: int, q: List[List[int]]) -> List[int]:
        ans = [0]
        x = 0
        for op, val in q:
            if op == 0:
                ans.append(val ^ x)
            else:
                x ^= val
        for i in range(len(ans)):
            ans[i] ^= x
        ans.sort()
        return ans

File: 02-06-2024_Construct_list_using_given_q_XOR_queries.cpp
class Solution {
  public:
    vector<int> constructList(int k, vector<vector<int>> &q) {
        vector<int> ans;
        int x = 0;
        ans.push_back(0);
        for(int i = 0 ; i<q.size() ; i++)
        {
            if(q[i][0]==0)
            {
                ans.push_back(q[i][1]^x);
            }
            else
            {
                x ^= q[i][1];
            }
        }
        for(int j = 0 ; j < ans.size() ; j++)
        {
            ans[j] ^= x;
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};
